src/Processors/Transforms/buildPushingToViewsChain.cpp

```diff
@@ -794,7 +794,7 @@ PushingToLiveViewSink::PushingToLiveViewSink(const Block & header, StorageLiveVi
 void PushingToLiveViewSink::consume(Chunk & chunk)
 {
     Progress local_progress(chunk.getNumRows(), chunk.bytes(), 0);
-    live_view.writeBlock(live_view, getHeader().cloneWithColumns(chunk.detachColumns()), std::move(chunk.getChunkInfos()), context);
+    live_view.writeBlock(live_view, getHeader().cloneWithColumns(chunk.getColumns()), std::move(chunk.getChunkInfos()), context);
```

Comment:
AFAIU as now in consume we always have a reference, while previously it used to be always movable object, then now we should not do detachColumns in consume ever?

Reply:
Yes. You do detach, dependant tables receive chunk without columns. In other hand, you could modify chunk in a way the dependant tables observe that.